co-simulation/fed/mosaic-carla/src/main/java/org/eclipse/mosaic/fed/carla/ambassador/CarlaAmbassador.java

```diff
@@ -349,32 +402,232 @@ public synchronized void processTimeAdvanceGrant(long time) throws InternalFeder
         }
 
         try {
-            if ( time == 0 ) {
-                // Try to connect to CARLA CDA Sim Adapter on first timestep
-                carlaXmlRpcClient.connect(60);
+            if (time == 0) {
+                // Try to connect to XML-RPC servers on first timestep
+                if (multiXmlRpcManager != null) {
+                    multiXmlRpcManager.connectAll(60);
+                } else if (carlaXmlRpcClient != null) {
+                    carlaXmlRpcClient.connect(60);
+                }
             }
             // if the simulation step received from CARLA, advance CARLA federate local
             // simulation time
-            if (isSimulationStep) {
-                List<DetectedObjectInteraction> detectedObjectInteractions = new ArrayList<>();
-                // Get all detections from all currently registered detectors.
-                for (DetectorRegistration registration: registeredDetectors ) {
-                    DetectedObject[] detections = carlaXmlRpcClient.getDetectedObjects( registration.getInfrastructureId() , registration.getDetector().getSensorId());
-                    for (DetectedObject detected: detections) {
-                        DetectedObjectInteraction interaction = new DetectedObjectInteraction(time, detected);
-                        // Convert nanosecond timestamp to millisecond timestamp
-                        interaction.getDetectedObject().setTimestamp((int)(time/1e6));
-                        detectedObjectInteractions.add(interaction);
+            // Advance CARLA simulation by one tick before polling sensors/actors
+            try {
+                CarlaXmlRpcClient tickClient = null;
+                if (multiXmlRpcManager != null) {
+                    tickClient = multiXmlRpcManager.getClient(CarlaXmlRpcClient.ServerType.ACTOR_LIB);
+                } else {
+                    tickClient = carlaXmlRpcClient;
```

Comment:
same question, what is carlaXmlRpcClient? if it is sensor xml rpc, I don't recall it is calling tick